README.md

```diff
@@ -73,55 +153,103 @@ USE_L10N = True
-```
-{% load transifex %}
 
+```django
+{% load transifex %}
 <p>{% t "Hello!" %}</p>
 <p>{% t "I want to be translated." %}</p>
 ```
 
-#### 2. Push source content to Transifex
+###### Django view simple example 
 
-This command will collect all translatable strings and push them to Transifex.
+```django
+from transifex.native.django import t
+from django.http import HttpResponse
+
+def my_view(request):
+    output = t("Welcome aboard!")
+    return HttpResponse(output)
 ```
+	
+To see more examples on translation hooks checkout [Use in Templates and Views](#use-in-templates-and-views).	
+
+##### Push Source Content to Transifex
+
+This command will collect all translatable strings and push them to Transifex.
+
+```bash
 ./manage.py transifex push
 ```
 
-#### 3. Translate content on Transifex
+##### Translate Content on Transifex
 
 The next step is for your translators to translate the strings in various languages using Transifex.
-When a translation is added on Transifex, it becomes available over-the-air on your app.
-Please note that it can take a few minutes for the translations to become available on your app.
+When a translation is added on Transifex, it becomes available over-the-air on your application.
+Please note that it can take a few minutes for the translations to become available on your application.
 
-#### 4. Display translated content
+##### Display Translated Content
 
 The Transifex Toolkit automatically displays translated content in the language currently selected in your Django project.
+In order to allow changing the current language, you will need a **language picker**.
 
-In order to allow changing the current language, you will need the following:
+Here is an example of how you can add a **language picker** in your application. 
 
-##### 4.1 A language picker
+###### Project preparation for localization
 
-Here is an example of how you can add a language picker in your app.
-You can add this on the same HTML file you added the translatable strings before, like so:
+Please follow this steps in order to prepare your Django project for localization:
+
+1. You can add this on the same HTML file you added the translatable strings before, like so:
+
+```html
+   <form>
+   <!-- ... -!>
+```
+
+2. Add the following route in your Project's routes, so that the `set_language` hook shown above will work when submitting the form.
+
+```django
+from django.conf.urls import url, include
+
+urlpatterns = [
+    ...,
+    url(r'^i18n/', include('django.conf.urls.i18n')),
+]
+```
+
+3. Last, add `'django.middleware.locale.LocaleMiddleware'` in your `settings.MIDDLEWARE` to enable the functionality.
+
+> Please refer to Django documentation for further information:
+>
+> https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#the-set-language-redirect-view
+> https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#how-django-discovers-language-preference
```

Comment:
I believe this would look better:

```
Please refer to Django documentation for further information:

> - https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#the-set-language-redirect-view
> - https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#how-django-discovers-language-preference 
```

Reply:
Done, looks more clear now